### PR TITLE
[#1135] Approval List v2 API

### PR DESCRIPTION
**Pull request by dealako**
- Added Logic to Send Emails to CLA Managers
- Added Logic to Send Emails to Contributors when email or GH Username is added/removed from the Approval List
- Updated Functional Tests to use API-GW/ACS endpoint
- Added additional functional test users
- Added Get user by Email service/db function
- Added Logic to create events when Approval list is updated

Signed-off-by: David Deal <dealako@gmail.com>

### CODE REVIEW

File: cla-backend-go/signatures/service.go [hunk 1]
```diff
@@ -375,3 +432,368 @@ func (s service) AddCLAManager(signatureID, claManagerID string) (*models.Signat
...
+	for _, value := range approvalList.AddEmailApprovalList {
+		// Send an event
+		s.eventsService.LogEvent(&events.LogEventArgs{
+			EventType:         events.ClaApprovalListUpdated,
+			ProjectID:         projectModel.ProjectID,
+			ProjectModel:      projectModel,
+			CompanyID:         companyModel.CompanyID,
+			CompanyModel:      companyModel,
+			LfUsername:        authUser.UserName,
+			UserID:            authUser.UserName,
```
Comment (prasannamahajan) [inline cla-backend-go/signatures/service.go:590]: This should be Users uuid

File: cla-backend-go/users/repository.go [hunk 2]
```diff
@@ -458,6 +460,110 @@ func (repo repository) GetUserByUserName(userName string, fullMatch bool) (*mode
...
+	// The user model
+	var dbUserModels []DBUser
+
+	err = dynamodbattribute.UnmarshalListOfMaps(result.Items, &dbUserModels)
+	if err != nil {
+		log.Warnf("error unmarshalling user record from database for user_github_username: %s, error: %+v", gitHubUsername, err)
+		return nil, err
+	}
+
+	if len(dbUserModels) == 0 {
```
Comment (prasannamahajan) [inline cla-backend-go/users/repository.go:558]: If we don't found user then we should return error 
ErrUserNotFound = "user not found"

It is common pattern. Many time we don't check if pointer is non nil or not,  if err is nil.

**Comment by prasannamahajan** [comment]
LGTM

File: cla-backend-go/cmd/functional_tests/approval_list/approval_list.go [hunk 3]
```diff
@@ -33,32 +29,129 @@ const (
...
 	claProspectiveManagerEmail = "ddeal+cla+dev+functional+test+user@linuxfoundation.org"
-	claProjectSFID             = "a0941000005ouJFAAY"
-	claManagerCompanySFID      = "external-ee965ea2-ca83-4482-8a1b-94a468d9dcfa"
-	claGroupID                 = "d5412846-5dda-4c58-8f62-4c111a3cd0d3"
-	testGitHubUsername         = "dealako"
-	testDomainName             = "dealako.com"
-	testGitHubOrg              = "deal-test-org"
+
+	projectSFID = "a092M00001If9v8QAB"
+	intelSFID   = "0014100000TdzA7AAJ"
```
Comment (wanyaland) [inline cla-backend-go/cmd/functional_tests/approval_list/approval_list.go:34]: the intelSFID = "014100000TdzA7AAJ" is not associated with intel and should be "00117000015vpjXAAQ". Discovered this issue on troubleshooting with Nirav

**Pull request merged**